src/Common/src/CoreLib/System/Text/ValueStringBuilder.cs

```diff
@@ -61,6 +68,12 @@ public bool TryCopyTo(Span<char> destination, out int charsWritten)
             }
         }
 
+        public void WriteTo(TextWriter writer)
```

Comment:
This method is very specialized. It should rather be `.AsReadOnlySpan()` property. It will handle all future uses that need to get the string as Span.

Reply:
Are you sure? That sounds like it could be a bit risky: when the underlying array gets resized the old one gets returned to the pool (but the `ReadOnlySpan` will still have a reference to it).

Reply:
This is internal type, not a public API. There are many ways you can use it in a wrong way. For example, you can accidentally pass it by value and create a second copy of it to introduce the same problem as if you use `AsReadOnlySpan` incorrectly by accident.

Reply:
[Done](https://github.com/dotnet/corefx/pull/27250/commits/ef33e71833b6b9d2ab03ed2c97cf3199e9bcf857)